api/src/services/taxonomy-service.ts

```diff
@@ -7,7 +7,7 @@ const defaultLog = getLogger('services/taxonomy-service');
 
 export type TaxonSearchResult = {
   tsn: number;
-  commonName: string | null;
+  commonNames: string[] | [];
```

Comment:
You should remove the `[]` type. This should just just read `commonNames: string[];`.